src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -277,6 +277,99 @@ void GraphOptimizer::FuseConvMatmulFCDeconvAndDQScales(Graph &graph) {
     }
 }
 
+void GraphOptimizer::FuseFCAndWeightsDecompression(Graph &graph) {
+    auto expectedNode = [](NodePtr node, Type expectedType) {
+        return node->getType() == expectedType && node->getChildEdges().size() == 1;
+    };
+
+    auto& graphNodes = graph.GetNodes();
+    for (size_t i = 0; i < graphNodes.size(); i++) {
+        const auto fcNode = dynamic_cast<node::FullyConnected*>(graphNodes[i].get());
+        if (fcNode == nullptr)
+            continue;
+
+        const auto parent = fcNode->getParentEdgesAtPort(1)[0]->getParent();
+        const bool with_transpose = parent->getType() == Type::Transpose;
+        const NodePtr transposeNode = with_transpose ? parent : nullptr;
+
+        const auto multiplyNode = with_transpose ? parent->getParentEdgesAtPort(0)[0]->getParent() : parent;
+        if (!expectedNode(multiplyNode, Type::Eltwise) || multiplyNode->getAlgorithm() != Algorithm::EltwiseMultiply ||
+            !multiplyNode->isConstant())
+            continue;
+
+        const auto multiplyConstNode = multiplyNode->getParentEdgesAtPort(1)[0]->getParent();
+        if (!expectedNode(multiplyConstNode, Type::Input))
+            continue;
+
+        const auto mulParent = multiplyNode->getParentEdgesAtPort(0)[0]->getParent();
+        const bool with_subtract = mulParent->getAlgorithm() == Algorithm::EltwiseSubtract;
+        NodePtr subtractNode, subtractConstNode;
+        if (with_subtract) {
+            subtractNode = mulParent;
+            if (!expectedNode(subtractNode, Type::Eltwise))
+                continue;
+            subtractConstNode = subtractNode->getParentEdgesAtPort(1)[0]->getParent();
+            if (!expectedNode(subtractConstNode, Type::Input))
+                continue;
+        }
+
+        const auto convertNode = with_subtract ? subtractNode->getParentEdgesAtPort(0)[0]->getParent() : mulParent;
+        if (!expectedNode(convertNode, Type::Convert))
+            continue;
+        const auto weightsNode = convertNode->getParentEdgesAtPort(0)[0]->getParent();
+        if (!expectedNode(weightsNode, Type::Input))
+            continue;
+
+        // Precision limitations
+        if (multiplyConstNode->getOriginalOutputPrecisionAtPort(0) != Precision::FP32)
+            continue;
+        if (weightsNode->getOriginalOutputPrecisionAtPort(0) != Precision::U8)
```

Comment:
Can you add vector with supported weights precision in the beggining of the pass and check it here? 
The expection is list of supported weights precision will be expanded in nearest future, so to minimize required changes for the developers it is reasonable to implement it as parameter.

Reply:
Added is `supportedWeightsPrecisions`. But as std::set, not std::vector. In my opinion, this is a more appropriate structure to describe the supported weights precisions